lib/std/zig/system.zig

```diff
@@ -221,7 +222,44 @@ pub const NativeTargetInfo = struct {
                     }
                 },
                 .windows => {
-                    // TODO Detect native operating system version.
+                    var version_info: std.os.windows.RTL_OSVERSIONINFOW = undefined;
+                    version_info.dwOSVersionInfoSize = @sizeOf(@TypeOf(version_info));
+
+                    const rc = std.os.windows.ntdll.RtlGetVersion(&version_info);
+                    switch (rc) {
+                        .SUCCESS => {},
+                        else => return std.os.windows.unexpectedStatus(rc),
+                    }
+
+                    // Starting from the system infos build a NTDDI-like version
+                    // constant whose format is:
+                    //   B0 B1 B2 B3
+                    //   `---` `` ``--> Sub-version (Starting from Windows 10 onwards)
+                    //     \    `--> Service pack (Always zero in the constants defined)
+                    //      `--> OS version (Major & minor)
+                    const os_ver: u16 = //
+                        @intCast(u16, version_info.dwMajorVersion & 0xff) << 8 |
+                        @intCast(u16, version_info.dwMinorVersion & 0xff);
+                    const sp_ver: u8 = 0;
+                    const sub_ver: u8 = if (os_ver >= 0xA000) subver: {
```

Comment:
This whole `if` might be better if it was only for `os_ver == 0xA000`: build 15063 is unlikely to be version 3 of windows 11

Reply:
Hm? The build check is already gated by the ` if (os_ver >= 0xA000)` check

Reply:
e.g. if `os_ver == 0xB000`: does `15063` imply 3 still?

Reply:
No idea, the headers declare `0xa000` as a "threshold value" `(NTDDI_WINTHRESHOLD)`, that's why I've used a `>=`. But we'll never see a Windows 11 so I think either way this is fine